File: src/main/java/io/vproxy/vfx/animation/AnimationApplyFunction.java
package io.vproxy.vfx.animation;

import io.vproxy.vfx.util.algebradata.AlgebraData;

@FunctionalInterface
public interface AnimationApplyFunction<T extends AlgebraData<T>> {
    void apply(AnimationNode<T> from, AnimationNode<T> to, T data);
}

File: src-ts/src/animation/AnimationApplyFunction.ts
// Plain function type (not interface with `apply` method) because `apply`
// collides with `Function.prototype.apply` — raw arrow functions would
// silently invoke the prototype method instead of the user's callback.
import type { AlgebraData } from '../util/algebradata/AlgebraData.js';
import type { AnimationNode } from './AnimationNode.js';

export type AnimationApplyFunction<T extends AlgebraData<T>> = (
  from: AnimationNode<T> | null,
  to: AnimationNode<T>,
  data: T,
) => void;
